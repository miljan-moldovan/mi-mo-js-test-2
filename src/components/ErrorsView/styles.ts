import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  errorIcon: {
    position: 'absolute',
    zIndex: 300,
    top: 13,
    left: 15,
    width: 12,
    height: 12,
    fontSize: 12,
    alignSelf: 'center',
    resizeMode: 'contain',
    color: 'rgba(209, 36, 41, 1)',
  },
  errorMessage: {
    marginLeft: 'auto',
    marginRight: 'auto',
    color: 'black',
    fontSize: 12,
    fontFamily: 'Roboto-Regular',
  },
  errorContainer: {
    // width: 292,
    flexDirection: 'row',
    alignItems: 'flex-start',
    marginLeft: 'auto',
    marginRight: 'auto',
    paddingLeft: 30,
    paddingRight: 20,
    marginTop: 2,
    justifyContent: 'space-between',
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    width: 311,
    borderRadius: 4,
  },
  errorItemsContainer: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-around',
    marginLeft: 15,
    marginBottom: 10,
    marginTop: 10,
  },
  errorListItem: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  errorListMessage: {
    fontSize: 11,
    fontFamily: 'Roboto-Regular',
    color: 'rgb(46, 49, 66)',
  },
  bullet: {
    width: 10,
  },
});
