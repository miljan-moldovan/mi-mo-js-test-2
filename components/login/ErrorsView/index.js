import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome, { Icons } from 'react-native-fontawesome';
import { Text, View } from 'react-native';

import styles from './styles';

function ErrorsView(props) {
  return (
    <View style={styles.errorContainer}>
      <FontAwesome style={styles.errorIcon}>{Icons.exclamationTriangle}</FontAwesome>
      {props.errors.length > 0 && (
        <View style={styles.errorItemsContainer}>
          {props.errors.map((err, index) => (
            <View style={styles.errorListItem}>
              <Text style={styles.bullet}>{'\u2022 '}</Text>
              <Text key={`err__${index}`} style={styles.errorListMessage}>{err}</Text>
            </View>
          ))}
        </View>
      )}
      {props.errors.length === 0 && (
        <Text style={styles.errorMessage}>{props.error}</Text>
      )}
    </View>
  );
}

ErrorsView.defaultProps = {
  error: '',
  errors: [],
};

ErrorsView.propTypes = {
  error: PropTypes.string,
  errors: PropTypes.arrayOf(PropTypes.string),
};

export default ErrorsView;
