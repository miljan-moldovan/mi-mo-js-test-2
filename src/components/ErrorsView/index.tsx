import * as React from 'react';
import PropTypes from 'prop-types';
import FontAwesome, { Icons } from 'react-native-fontawesome';
import { Text, View } from 'react-native';

import styles from './styles';

function ErrorsView(props) {
  return (
    <View style={styles.errorContainer}>
      <FontAwesome style={styles.errorIcon}>{Icons.exclamationTriangle}</FontAwesome>
      <View style={styles.errorItemsContainer}>
        {props.errors.length > 0 && (
          props.errors.map((err, index) => (
            <View style={styles.errorListItem}>
              <Text style={styles.bullet}>{'\u2022 '}</Text>
              <Text key={`err__${index}`} style={styles.errorListMessage}>{err}</Text>
            </View>
          ))
        )}
        {props.errors.length === 0 && props.error && (
          <View style={styles.errorListItem}>
            <Text style={styles.bullet}>{'\u2022 '}</Text>
            <Text key="err_0" style={styles.errorListMessage}>{props.error}</Text>
          </View>
        )}
      </View>
    </View>
  );
}

ErrorsView.defaultProps = {
  error: '',
  errors: [],
};

ErrorsView.propTypes = {
  error: PropTypes.string,
  errors: PropTypes.arrayOf(PropTypes.string),
};

export default ErrorsView;
