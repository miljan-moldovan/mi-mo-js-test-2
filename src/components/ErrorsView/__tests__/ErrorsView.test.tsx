import * as React from 'react';
import { shallow, mount } from 'enzyme';

import ErrorsView from '../';

describe('ErrorView', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ErrorsView/>,
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
