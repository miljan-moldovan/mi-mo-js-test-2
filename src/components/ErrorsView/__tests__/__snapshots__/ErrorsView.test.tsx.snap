// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "rgba(255, 255, 255, 0.5)",
      "borderRadius": 4,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 2,
      "paddingLeft": 30,
      "paddingRight": 20,
      "width": 311,
    }
  }
>
  <Icon
    style={
      Object {
        "alignSelf": "center",
        "color": "rgba(209, 36, 41, 1)",
        "fontSize": 12,
        "height": 12,
        "left": 15,
        "position": "absolute",
        "resizeMode": "contain",
        "top": 13,
        "width": 12,
        "zIndex": 300,
      }
    }
  >
    ÔÅ±
  </Icon>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-around",
        "marginBottom": 10,
        "marginLeft": 15,
        "marginTop": 10,
      }
    }
  />
</View>
`;
